ecs.tf

```diff
@@ -91,6 +91,41 @@ resource "aws_iam_role_policy" "s3_data_for_ec2" {
 EOF
 }
 
+resource "aws_iam_role_policy" "external_s3_data_for_ec2" {
```

Comment:
It might be more compact to list all the buckets in one policy than to create a policy for every bucket. The limit on inline policies is 10240 chars in aggregate.

But I'm fine with using this as-is for now, and then we can refactor it later to do that.

Reply:
this is configured to use "count", so will add separate entry for each bucket.
name  = "external-s3-data-for-ec2-${count.index}"

Reply:
What I understand is there will be a separate named inline policy for each bucket, with repetition of the content of the policy (apart from the bucket name itself). If we have 3 buckets we will have 3 inline policies, with names: external-s3-data-for-ec2-0, external-s3-data-for-ec2-1, and external-s3-data-for-ec2-2

The content of each of those policies is mostly the same, only the bucket name changes. The overall limit on all inline policies in a single role, in aggregate, is 10240 characters, so the repetition of the policy template part will count against those 10240 characters in each instance of the policy.

I think it's fine for now and I would just merge this as-is, just pointing out that a more compact representation would be to have a single inline policy with all the buckets in it in a single statement. That will be less character repetition and the overall inline policy character count would be shorter.

Reply:
ok, I didn't know that limitation, yes we can refactor to make it optimal.

Reply:
We can refactor later to pass the bucket names via variable and put in the policy